pyts/metrics/dtw.py

```diff
@@ -1197,28 +1323,53 @@ def dtw(x=None, y=None, dist='square', method='classic', options=None,
-        'window_size' becomes a valid cell for the path.
+    Notes
+    -----
+    This section describes the available versions of Dynamic Time Warping (DTW)
+    that can be selected by the `method` parameter.
 
-    max_slope : float (default = 2.)
-        Maximum slope for the parallelogram.
+    **Unconstrained path**
 
-    resolution : int (default = 2)
-        The resolution level.
+    Method :ref:`'classic'<metrics.dtw-classic>` computes the original DTW
+    score between two time series with no constraint region: each cell is a
+    valid cell to find the optimal path minimizing the total cost.
 
-    radius : int (default = 0)
-        The radius used to expand the constraint region. The optimal path
-        computed at the resolution level is expanded with `radius` cells to the
-        top, bottom, left and right of every cell belonging to the optimal
-        path. It is computed at the resolution level.
+    **Global constraint region**
+
+    Using no constraint region allows for very large distortion between both
+    time series, which may be ill-suited for some cases. One possible solution
+    to this issue is to use a global constraint region for the optimal path.
+    A cell can be part of the optimal path if and only if this cell is in the
+    constraint region. Another advantage of using a global constraint region is
+    to decrease the computational complexity to find the optimal path.
+
+    Method :ref:`'sakoechiba'<metrics.dtw-sakoechiba>` uses a band as the
+    constraint region, allowing for a maximum fixed time shift at every time
+    point.
+
+    Method :ref:`'itakura'<metrics.dtw-itakura>` uses a parallelogram as
+    the constraint region, allowing for time shifts of varying length: small
+    time shifts at the beginning and at the end of the time series, larger
+    time shifts in the middle.
+
+    **Adaptative constraint region**
+
+    One drawback of global constraint regions is that they do not depend on
+    the time series and might be ill-suited for some time series. Adaptative
+    constraint regions are regions that are specific to the two provided
+    time series. However, their computational complexities are larger that
+    the ones of global regions.
+
+    Method :ref:`'multiscale'<metrics.dtw-multiscale>` downsamples both time
+    series by a given factor, computes the optimal path with no constraint
+    region for the downsampled time series, and projects the optimal path on
+    the original scale to define the constraint region.
+
+    Method :ref:`'fast'<metrics.dtw-fast>` can be seen as a recursive version
+    of 'multiscale': the time series are downsampled so that the new time
+    series are very small, the optimal path for these time series is computed
+    and is projected at the scale that is two times larger. This process is
+    repeated until the original scale of the time series is reached.
```

Comment:
These descriptions are better than the ones provided in the individual docstrings. Maybe it's worth replacing them with these.

Reply:
It's easier because we can talk about all the methods and how they relate to each other, while the individual docstrings need to be independent.

It's quite similar to [scipy.optimize.minimize](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html): the `Notes` section is very detailed, while the docstring for each method (e.g. [minimize(method='Nelder-Mead')](https://docs.scipy.org/doc/scipy/reference/optimize.minimize-neldermead.html#optimize-minimize-neldermead)) is very short and does not give much information.